InjectionIII/AppDelegate.swift

```diff
@@ -0,0 +1,218 @@
+    }
+
+    @IBAction func toggleVaccine(_ sender: NSMenuItem) {
+        toggleState(sender)
+        let newSetting = sender.state == NSControl.StateValue.on
+        UserDefaults.standard.set(newSetting, forKey:UserDefaultsVaccineEnabled)
+        self.lastConnection?.sendCommand(.vaccineSettingChanged, with:vaccineConfiguration())
+    }
+
+    @IBAction func traceApp(_ sender: NSMenuItem) {
+        toggleState(sender)
+        self.lastConnection?.sendCommand(sender.state == NSControl.StateValue.on ?
+            .trace : .untrace, with: nil)
+    }
+
+    func vaccineConfiguration() -> String {
+        let vaccineSetting = UserDefaults.standard.bool(forKey: UserDefaultsVaccineEnabled)
+        let dictionary = [UserDefaultsVaccineEnabled: vaccineSetting]
+        let jsonData = try! JSONSerialization
+            .data(withJSONObject: dictionary, options:[])
+        let configuration = String(data: jsonData, encoding: .utf8)!
+        return configuration
+    }
+
+    func application(_ theApplication: NSApplication, openFile filename: String) -> Bool {
+        let open = NSOpenPanel()
+        open.prompt = NSLocalizedString("Select Project Directory", tableName: "Project Directory", comment: "Project Directory")
+        //    open.allowsMultipleSelection = TRUE;
+        if filename != "" {
+            open.directoryURL = URL(fileURLWithPath: filename)
+        }
+        open.canChooseDirectories = true
+        open.canChooseFiles = true
+        //    open.showsHiddenFiles = TRUE;
+        if open.runModal().rawValue == NSFileHandlingPanelOKButton {
+            let fileList = try? FileManager.default
+                .contentsOfDirectory(atPath: open.url!.path)
+            if let fileList = fileList, let projectFile =
+               fileWithExtension("xcworkspace", inFiles: fileList) ??
+               fileWithExtension("xcodeproj", inFiles: fileList),
+                let url = open.url {
+                self.selectedProject = url
+                    .appendingPathComponent(projectFile).path
+                self.watchedDirectories.removeAll()
+                self.watchedDirectories.insert(url.path)
+                self.lastConnection?.setProject(self.selectedProject!)
+                NSDocumentController.shared
+                    .noteNewRecentDocumentURL(url)
+                return true
+            }
+        }
+        return false
+    }
+
+    func fileWithExtension(_ ext: String, inFiles files: [String]) -> String? {
+        for file in files {
+            if (file as NSString).pathExtension == ext {
+                return file
+            }
+        }
+        return nil
+    }
+
+    func setMenuIcon(_ tiffName: String) {
+        DispatchQueue.main.async {
+            if let path = Bundle.main.path(forResource: tiffName, ofType:"tif"),
+                let image = NSImage(contentsOfFile: path) {
+    //            image.template = TRUE;
```

Comment:
We could set `isTemplate` to `true` if we want here.

Reply:
Not sure if this was related to dark mode, but I think you loose the colours